vm/systemSmartContracts/delegationManager.go

```diff
@@ -97,29 +113,262 @@ func (d *delegationManager) Execute(args *vmcommon.ContractCallInput) vmcommon.R
 		return d.init(args)
 	case "createNewDelegationContract":
 		return d.createNewDelegationContract(args)
-	case "deleteDelegationContract":
-		return d.deleteDelegationContract(args)
 	case "getAllContractAddresses":
 		return d.getAllContractAddresses(args)
+	case "changeBaseIssuingCost":
+		return d.changeBaseIssuingCost(args)
+	case "changeMinDeposit":
+		return d.changeMinDeposit(args)
 	}
 
+	d.eei.AddReturnMessage("invalid function to call")
 	return vmcommon.UserError
 }
 
-func (d *delegationManager) getAllContractAddresses(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+func (d *delegationManager) init(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if args.CallValue.Cmp(zero) != 0 {
```

Comment:
Maybe add a check as to not call init other from a hardcoded address? Something done in `checkConfigChangeInput` function

Reply:
no need. as this can be called only from RunSmartContractCreate - which in metachain can be called by the system only.

Reply:
...and because _init functions can be called only by internal code, as seen in systemVM.RunSmartContractCall function